bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -210,11 +167,35 @@ public ObjectId(final Date date, final int machineIdentifier, final short proces
      * @param processIdentifier the process identifier
      * @param counter           the counter
      * @throws IllegalArgumentException if the high order byte of machineIdentifier or counter is not zero
+     * @deprecated Use {@link #ObjectId(int, int)} instead
      */
+    @Deprecated
     public ObjectId(final int timestamp, final int machineIdentifier, final short processIdentifier, final int counter) {
         this(timestamp, machineIdentifier, processIdentifier, counter, true);
     }
 
+    /**
+     * Creates an ObjectId using the given time, machine identifier, process identifier, and counter.
+     *
+     * @param timestamp         the time in seconds
+     * @param counter           the counter
+     * @throws IllegalArgumentException if the high order byte of counter is not zero
+     */
+    public ObjectId(final int timestamp, final int counter) {
+        this(timestamp, counter, true);
+    }
+
+    private ObjectId(final int timestamp, final int counter, final boolean checkCounter) {
```

Comment:
Change to

    this(timestamp, RANDOM_VALUE1, RANDOM_VALUE2, counter, checkCounter);

?  Generally good for constructors to delegate to each other.

Reply:
I have made this change.